nce_cost/README_en.md

```diff
@@ -0,0 +1,161 @@
+| neg\_distribution        |  Generates the distribution of negative sample tags. The default is a uniform distribution| You can control the sampling weights for each category when negative samples are sampled. For example, if you want the positive example to be “sunny,” and the negative example “flood” is highlighted more during training, you can increase the weight of the “flood” sample |
+
+
+
+## Prediction
+1. Runing in command line :
+    ```bash
+    python infer.py \
+      --model_path "models/XX" \
+      --batch_size 1 \
+      --use_gpu false \
+      --trainer_count 1
+    ```
+    The parameters are as follows：
+    - `model_path`：Specify the path where the trained model is located. required.
+    - `batch_size`：The number of samples in parallel is predicted at one time. Optional, the default value is `1`.
+     - `use_gpu`：Whether use GPU for prediction。Optional，the default value is `False`。
+    - `trainer_count` : Number of thread for prediction。Optional，the default is  `1`。**Note: The number of threads used for prediction must be greater than the number of samples to be predicted in parallel**。
+
+
+2. It should be noted that: **The calculation logic for prediction and training is different**. Predictions using full-join matrix multiplication followed by `softmax` activation, output based on the probability distribution for each category, need to replace the `paddle.train.nce` layer used in training. In PaddlePaddle, the NCE layer stores the learnable parameters as a matrix of `[number of classes × output vector width of the previous layer]`. When predicting, **full-join operation needs to be transferred when loading the NCE layer to learn the parameters**, the code is as follows:
+    ```python
+    return paddle.layer.mixed(
+          size=dict_size,
+          input=paddle.layer.trans_full_matrix_projection(
+              hidden_layer, param_attr=paddle.attr.Param(name="nce_w")),
+          act=paddle.activation.Softmax(),
+          bias_attr=paddle.attr.Param(name="nce_b"))
+    ```
+    The `paddle.layer.mixed` in the above code snippet must be entered as `paddle.layer.×_projection` in PaddlePaddle. `paddle.layer.mixed` sums up multiple `projection` (input can be multiple) calculations as output. `paddle.layer.trans_full_matrix_projection` Transposes the parameter $W$ when calculating matrix multiplication.
+
+3. The forecasted output format is as follows:
+     ```text
+     0.6734  their   may want to move
+     ```
+
+     Each row is a forecast result, internally separated by "\t", for a total of 3 columns:
+     - First column: Probability of the next word.
+     - Second column: The next word of the model prediction.
+     - Third column: $n$ words entered, separated by spaces.
+
+
+## Reference
```

Comment:
## Prediction
1. Run from the command line window:
    ```bash
    Python infer.py \
      --model_path "models/XX" \
      --batch_size 1 \
      --use_gpu false \
      --trainer_count 1
    ```
    The parameters are as follows:
    - `model_path`: specifies the path where the trained model is located. This parameter is required.
    - `batch_size`: predicts the number of samples in parallel at a time. Optional, the default value is `1`.
    - `use_gpu`: determines whether or not to use GPU for prediction. Optional, the default value is `False`.
    - `trainer_count` : The number of threads to use for prediction. Optional, defaults to `1`. **Note: The number of threads used for prediction must be greater than the number of samples to be predicted concurrently**.

2. It should be noted that** the calculation logic for prediction and training is different**. During the prediction phase, the model uses full-joined matrix multiplication followed by `softmax` activation and outputs the probability distribution for each category. The model needs to replace the `paddle.train.nce` layer used in the training. In PaddlePaddle, the NCE layer stores the learnable parameters as a matrix of `[number of classes × output vector width of the previous layer]`. When predicting, **full-joined operations need to be transferred when loading the NCE layer to learn the parameters. Set **, the code is as follows:
    ```python
    Return paddle.layer.mixed(
          Size=dict_size,
          Input=paddle.layer.trans_full_matrix_projection(
              Hidden_layer, param_attr=paddle.attr.Param(name="nce_w")),
          Act=paddle.activation.Softmax(),
          Bias_attr=paddle.attr.Param(name="nce_b"))
    ```
     `paddle.layer.mixed` in the above code snippet must be entered as `paddle.layer.×_projection` in PaddlePaddle. `paddle.layer.mixed` sums up multiple `projection` (input can be multiple) calculations as output. `paddle.layer.trans_full_matrix_projection` transposes parameter $W$ when calculating matrix multiplication.

3. The predict output format is as follows:
    ```text
    0.6734 their may want to move
    ```

    Each row is a prediction result with a total of 3 columns separated by "\t":
    - First column: probability of the next word.
    - Second column: the next word of the model prediction.
    - Third column: $n$ words entered, separated by spaces.


## References